ai_services/anomaly_detection/data_preprocessing_examples/oci_data_flow_based_examples/example_code/ad_utils.py

```diff
@@ -59,9 +89,23 @@ def _create_model_(self, project_id, compartment_id, data_asset_id,
             model_training_details=model_training_details)
         create_model_response = self.ad_client.create_model(
             create_model_details)
+        print(f"Create model request received "
+              f"status [{create_model_response.status}] "
+              f"opc-request-id [{get_header_value(create_model_response.headers, OPC_REQUEST_ID_KEY)}]")
         assert create_model_response.status == 201, \
             f"Error creating model: {create_model_response.text}"
-        return create_model_response.data.id
+
+        model_id = create_model_response.data.id
+        work_request_id = get_header_value(create_model_response.headers, OPC_WORK_REQUEST_ID_KEY)
+
+        assert self._poll_work_request_(work_request_id) in WORK_REQUEST_SUCCESS_STATES, \
+            f"Unable to train model [{model_id}]!"
```

Comment:
This doesn't look right to me if I understand correctly.. the model training is unlikely to success immediately after the create model request is submitted

Reply:
By discussing with Ravi offline, @backoff.on_exception(...) (will retry exponentially with backoff on a few defined errors and @backoff.on_predicate(...) will retry after 2 mins whenever the states are not in terminal states. So, the control will not go to the assert statement immediately. 

This is a blocking call and it won't return the result until the model training is done. So I feel my question is answered.